angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts

```diff
@@ -16,105 +17,197 @@ export class MinesAddEditComponent implements OnInit, OnDestroy {
   private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
 
   public loading = true;
-  public isEditing = false;
-  public record;
+  public mine: Mine;
   public myForm: FormGroup;
   public lastEditedSubText = null;
 
-  // Flavour data
-  public nrcedFlavour = null;
-  public lngFlavour = null;
-  public lngPublishSubtext = 'Not published';
-  public nrcedPublishSubtext = 'Not published';
-
-  // mine types
-  public types = ['Coal', 'Metal', 'Industrial Mineral', 'Sand & Gravel'];
+  public mineTypes = Picklists.mineTypes;
 
   constructor(
     public route: ActivatedRoute,
     public router: Router,
     private factoryService: FactoryService,
     private loadingScreenService: LoadingScreenService,
+    private utils: Utils,
     private _changeDetectionRef: ChangeDetectorRef
   ) {}
 
   ngOnInit() {
+    this.loadingScreenService.setLoadingState(true, 'main');
+
     this.route.data.pipe(takeUntil(this.ngUnsubscribe)).subscribe((res: any) => {
-      this.isEditing = res.breadcrumb !== 'Add Mine';
-      if (this.isEditing) {
-        if (res && res.record && res.record[0] && res.record[0].data) {
-          this.record = new Mine(res.record[0].data);
-        } else {
-          alert('Error: could not load edit mine.');
-          this.router.navigate(['mines']);
-        }
+      if (res && res.mine && res.mine[0] && res.mine[0].data) {
+        this.mine = new Mine(res.mine[0].data);
       } else {
-        // Add
-        if (res && res.record) {
-          this.record = new Mine(res.record);
-        }
+        alert('Error: could not load edit mine.');
+        this.router.navigate(['mines']);
       }
+
       this.buildForm();
+
       this.loading = false;
+      this.loadingScreenService.setLoadingState(false, 'main');
+
       this._changeDetectionRef.detectChanges();
     });
   }
 
-  private buildForm() {
+  /**
+   * Derive static text strings.
+   *
+   * @memberof MinesAddEditComponent
+   */
+  populateTextFields() {
+    if (this.mine.dateUpdated) {
```

Comment:
Do we need a null check on the mine here?

Reply:
yup, will add